src/subgraph/dex/client.ts

```diff
@@ -0,0 +1,24 @@
+import { Maybe, TokenInfo } from "../../types";
+import { getLogger } from "../../utilities";
+
+import * as actions from "./actions";
+import { createApolloClient } from "../helpers";
+
+const logger = getLogger("subgraph:dexClient");
+
+export interface DexSubgraphClient {
+  getTokenInfo(tokenId: string): Promise<Maybe<TokenInfo>>;
+}
+
+export const createDexClient = (dexSubgraphUri: string): DexSubgraphClient => {
```

Comment:
This should probably be called "UniswapClient" or something, since this is specifically for Uniswap and not any generic Dex